components/expenses/ExpenseForm.js

```diff
@@ -349,6 +349,18 @@ const ExpenseFormBody = ({
     }
   }, [values.payeeLocation]);
 
+  React.useEffect(() => {
+    if (
+      values.currency &&
+      values.payoutMethod?.data?.currency &&
+      values.payoutMethod.data.currency !== values.currency
+    ) {
+      const items = values.items.map(item => ({ ...item, amount: null }));
```

Comment:
As shared in https://github.com/opencollective/opencollective-frontend/pull/7530#issuecomment-1058092806, I find the currency reset a bit aggressive. At the time I proposed [this approach](https://github.com/opencollective/opencollective-frontend/pull/7530#issuecomment-1058092806):
![image](https://user-images.githubusercontent.com/1556356/214351359-5fe3bd1e-c840-446d-a790-1a35f271226b.png)

Not a blocker for merging, we should now prioritize fixing the issue and come back for the UX if needed.

Reply:
@Betree, I agree it is aggressive. But right now, it is the only behavior that requires explicit user input.

Reply:
@Betree : Yes it's ready

> This resolves [opencollective/opencollective#5206](https://github.com/opencollective/opencollective/issues/5206). We should also close #7530 if we merge this.
> 
> @SudharakaP Any blocker on your side? #7530 has become really huge and I don't have the context in mind anymore. Is it ready?

@Betree : Yes it's ready. I've addressed all the review comments. Just did a rebased from master as well so we don't have conflicts.